src/localStorage.ts

```diff
@@ -0,0 +1,24 @@
+export const loadState = () => {
+  try {
+    const localAllTasksState: string | null = localStorage.getItem(
+      "reduxState"
+    );
+
+    if (!localAllTasksState) return undefined;
+    // when use combinedReducers()
+    // preload state must be a plain object with the same shape as the keys passed to it
+    return { allTasksReducer: JSON.parse(localAllTasksState) };
+  } catch (error) {
+    // if error, return undefined to let reducer initialize app
+    return undefined;
```

Comment:
Just personal preference/something I have been told: for cases like this, where something is left intentionally "empty", you should use `null` instead of `undefined`.

Reply:
I followed a tutorial to undefined. I think that this is intentional(maybe), see below paragraph from redux docs. I wonder if null would still work with the createStore()...

"There are two main ways to initialize state for your application. The createStore method can accept an optional preloadedState value as its second argument. Reducers can also specify an initial value by looking for an incoming state argument that is undefined, and returning the value they'd like to use as a default."

"When a store is created, Redux dispatches a dummy action to your reducer to populate the store with the initial state. You are not meant to handle the dummy action directly. Just remember that your reducer should return some kind of initial state if the state given to it as the first argument is undefined, and you're all set."

here are the links: https://redux.js.org/recipes/structuring-reducers/initializing-state
https://redux.js.org/api/createstore